cocotb/regression.py

```diff
@@ -617,21 +618,21 @@ def _create_test(function, name, documentation, mod, *args, **kwargs):
     Returns:
         Decorated test function
     """
+    # causes early type check, during call to generate_tests()
+    test = cocotb.scheduler.create_task(function(cocotb.top, *args, **kwargs))
+
+    @wraps(function)
     async def _my_test(dut):
-        await function(dut, *args, **kwargs)
+        await test
 
-    _my_test.__name__ = name
-    _my_test.__qualname__ = name
-    _my_test.__doc__ = documentation
-    _my_test.__module__ = mod.__name__
     return cocotb.test()(_my_test)
 
 
 class TestFactory:
     """Factory to automatically generate tests.
 
     Args:
-        test_function: The function that executes a test.
+        test_function: A callable that return the test Coroutine.
```

Comment:
```suggestion
        test_function: A callable that returns the test Coroutine.
```
Or maybe even an uppercase `Callable`?